crates/handler/src/frame.rs

```diff
@@ -153,6 +153,7 @@ where
 
         // ExtDelegateCall is not allowed to call non-EOF contracts.
         if is_ext_delegate_call && !bytecode.bytes_slice().starts_with(&EOF_MAGIC_BYTES) {
+            context.journal().checkpoint_revert(checkpoint);
```

Comment:
This was causing the depth field to read incorrectly when this case triggered.  Not sure if here was any way to leverage it.  Also, EOF only.

Reply:
Good find! Yeah it is good fix and it is EOF only